### PR TITLE
STM32: Board conversion to DT clock configuration. Mix of Tested boards.

### PR DESCRIPTION
**Pull request by ABOSTM**
Following #34609, Board conversion to DT clock configuration.
All boards have been tested with samples/hello_world.
* nucleo_l152re
* nucleo_wl55jc
* nucleo_f207zg
* nucleo_f756zg
* nucleo_f767zi
* nucleo_l073rz
* nucleo_f030r8
* nucleo_f091rc
* nucleo_g474re
* nucleo_f303k8
* nucleo_l031k6
* stm32f3_disco
* b_l4s5i_iot01a

### CODE REVIEW

File: boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts [hunk 1]
```diff
@@ -46,6 +46,28 @@
 	};
 };
 
+&clk_hsi {
+	clock-frequency = <DT_FREQ_M(16)>;
```
[the remark this answers is not in the document]
Comment (ABOSTM) [reply boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts]: Right. Fixed for all boards configured with HSI.
Thanks

File: boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig [hunk 2]
```diff
@@ -30,19 +27,5 @@ CONFIG_PINMUX=y
 # GPIO Controller
 CONFIG_GPIO=y
 
-# Clock configuration
+# Clock # Enable Clocks
```
[the remark this answers is not in the document]
Comment (ABOSTM) [reply boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig]: Fixed. Thanks

**Review by erwango — APPROVED** [review]

File: boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts (hunk 1 above)
[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts]: Thanks @tagunil for you careful reviews

File: boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts [hunk 3]
```diff
@@ -46,6 +46,27 @@
...
+&clk_hsi {
+	status = "okay";
+};
+
+&pll {
+	div-m = <4>;
+	mul-n = <40>;
+	div-q = <2>;
+	div-r = <2>;
+	clocks = <&clk_hsi>;
```
Comment (FRASTM) [inline boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts:58]: Is `div-p` from  "CONFIG_CLOCK_STM32_PLL_P_DIVISOR=2", ignored

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**